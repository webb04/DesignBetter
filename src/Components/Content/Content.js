import React from 'react';
import PropTypes from 'prop-types';
import styled, { keyframes } from 'styled-components';
import { fadeInUp } from 'react-animations';

const fadeAnimation = keyframes`${fadeInUp}`;

const Section = styled.section`
    padding: 4em 7px;
`;

const Feature = styled.section`
    width: ${props => `calc(100% / ${props.number})`};
    max-width: 33%;
    display: inline-flex;
    align-items: center;
    justify-content: center;
    flex-direction: column;
    margin-bottom: 4em;
    animation: .4s ${fadeAnimation};
    border: none;
`;

const Image = styled.img`
    width: 50%;
    height: 50%;
    text-align: center;
`;

const FeatureTitle = styled.div`
    margin: 10px 0;
`;

const FeatureSubtitle = styled.div`
    width: 60%;
    font-weight: 300;
    font-size: 14px;
    text-align: center;
    color: #495367;
`;

const Content = ({ features }) => (
    <Section>
        {
        	features.map(({ image, title, subtitle }, i) => {
        		return (
          			<Feature className="feature" number={3} key={`feature-${i}`}>
            			<Image src={`/Images/${image}`}/>
            			<FeatureTitle>{title}</FeatureTitle>
            			<FeatureSubtitle>{subtitle}</FeatureSubtitle>
          			</Feature>
        		)
      		})
    	}
  	</Section>
)

Content.propTypes = {
    features: PropTypes.arrayOf(PropTypes.shape({
        image: PropTypes.string.isRequired,
        title: PropTypes.string.isRequired,
        subtitle: PropTypes.string.isRequired
    })).isRequired
}

export default Content;