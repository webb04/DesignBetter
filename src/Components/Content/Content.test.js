import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Content from './Content';
import { shallow } from 'enzyme';

const features = [
    {
        image: 'cloud/prototyping.png',
        title: 'Design Tool',
        subtitle: 'Share every design'
    },
    {
        image: 'cloud/prototyping.png',
        title: 'Design Tool',
        subtitle: 'Share every design'
    }
];

it('Renders the correct number of features', () => {
    const content = shallow(<Content features={features}/>);
    expect(content.find('.feature')).toHaveLength(2);
});

it('renders correctly', () => {
  const tree = renderer
    .create(<Content features={features}/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});