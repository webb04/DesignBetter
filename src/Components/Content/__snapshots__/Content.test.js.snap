// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section
  className="sc-bdVaJa rCoSi"
>
  <section
    className="feature sc-bwzfXH drZmNJ"
  >
    <img
      className="sc-htpNat eUYtJk"
      src="/Images/cloud/prototyping.png"
    />
    <div
      className="sc-bxivhb kWEhey"
    >
      Design Tool
    </div>
    <div
      className="sc-ifAKCX hhjABF"
    >
      Share every design
    </div>
  </section>
  <section
    className="feature sc-bwzfXH drZmNJ"
  >
    <img
      className="sc-htpNat eUYtJk"
      src="/Images/cloud/prototyping.png"
    />
    <div
      className="sc-bxivhb kWEhey"
    >
      Design Tool
    </div>
    <div
      className="sc-ifAKCX hhjABF"
    >
      Share every design
    </div>
  </section>
</section>
`;
